test/distributed/org/apache/cassandra/distributed/test/ring/BootstrapTest.java

```diff
@@ -110,15 +112,31 @@ public void hostIdOverrideTest() throws Throwable
             populate(cluster,0, 100);
             IInstanceConfig config = cluster.newInstanceConfig();
             IInvokableInstance newInstance = cluster.bootstrap(config);
+
+            
             withProperty("cassandra.host_id_first_boot", "06fc931f-33b5-4e22-0001-000000000001",
                          () -> newInstance.startup(cluster));
 
             newInstance.shutdown().get();
             newInstance.startup();
             String host_id = newInstance.executeInternal("SELECT host_id FROM system.local")[0][0].toString();
             Assert.assertEquals(host_id, "06fc931f-33b5-4e22-0001-000000000001");
-    	}
-    	
+            
+            // stop and run with another host id, check if value is being cached
+            newInstance.shutdown().get();
+            withProperty("cassandra.host_id_first_boot", "stoopid-host-id",
+                    () -> newInstance.startup());
+            host_id = newInstance.executeInternal("SELECT host_id FROM system.local")[0][0].toString();
+            Assert.assertEquals(host_id, "06fc931f-33b5-4e22-0001-000000000001");
+            
+            // check if another cluster member is aware of proper host id
+            IInvokableInstance member = cluster.get(1);
+            Object[][] result = member.executeInternal("SELECT host_id FROM system.peers");
```

Comment:
Why are you querying all peers here? You can restrict the query with `peer = newInstance.broadcastAddress()` and use `assertEquals`. You should also check the table `peers_v2` which is the new table for peers.

Reply:
done, now both tables are checked